client/e2e/pages/authenticated/routes/page.spec.ts

```diff
@@ -50,119 +58,121 @@ function buildAccessLists(currentRole: UserRole): AppRoute[] {
-              const profilePage = new ProfilePOM(page);
-              profilePage.urlIsCorrect();
+            case AppRoute.HOME:
+              pomPage = new HomePOM(page);
               break;
-            default:
-              // 👤 All other roles, routing results are based on route accesibility
-              if (isAllowedRoute) {
-                // 🔑 Accessible route, role has access
-                // 📌 Some routes have exceptions to the expected results
-                switch (route) {
-                  case AppRoute.HOME:
-                    // 👤 Authenticated users never get to home, redirected to dashboard
-                    // 🔍 Assert that the current URL ends with "/dashboard"
-                    const dashboardPage = new DashboardPOM(page);
-                    dashboardPage.urlIsCorrect();
-                    break;
-                  default:
-                    // 🛸 All other routes
-                    // 🔍 Assert that the current URL is correct
-                    await pomPage.urlIsCorrect();
-                    // 🔍 Assert that the not-found selector is not available
-                    // Wait for the selector to not be available
-                    await page.waitForSelector('[data-testid="not-found"]', {
-                      state: "hidden",
-                    });
-                    const notFoundSelector = await page.$(
-                      '[data-testid="not-found"]',
-                    );
-                    expect(notFoundSelector).toBeFalsy();
-                    break;
-                }
-              } else {
-                // 🔒 Inaccessible route, role has no access
-                // 📌 Some role && routes have exceptions to the expected results
-                if (
-                  value === UserRole.INDUSTRY_USER &&
-                  (route === AppRoute.OPERATION ||
-                    route === AppRoute.OPERATIONS)
-                ) {
-                  // 👤 Authenticated INDUSTRY_USER with 1 action pending get redirected to dashboard
+            case AppRoute.OPERATION:
+              pomPage = new OperationPOM(page);
+              break;
+            case AppRoute.OPERATIONS:
+              pomPage = new OperationsPOM(page);
+              break;
+            case AppRoute.OPERATOR:
+              pomPage = new OperatorPOM(page);
+              break;
+            case AppRoute.OPERATORS:
+              pomPage = new OperatorsPOM(page);
+              break;
+            case AppRoute.PROFILE:
+              pomPage = new ProfilePOM(page);
+              break;
+            case AppRoute.USERS:
+              pomPage = new UsersPOM(page);
+              break;
+          }
+          if (pomPage) {
+            // 🚨 Check if route is in the role's allow list
+            const isAllowedRoute = accessLists.includes(route);
+            // 🛸 Navigate to route
+            await pomPage.route();
+            // 📌 Some roles have exceptions to the expected results
```

Comment:
I found this test hard to follow--lots of exceptions to the defaults and different cases. If the test changes to only cover not-allowed routes though, I think that'll naturally improve readability.

Reply:
@BCerki  If you are suggesting to test only `not-allowed routes` I would suggest that in our implementation of the app route access it is easier to **not** have a route a role should have over having a route a role should not have. So, allow routes are important to test IMO.

Note: this test, with its exceptions,  already exists in develop, I have only re-arranged the looping so to get better details from the test descriptions

Reply:
I agree that allow routes are important to test--I was only suggesting removing them here because they're already going to be covered in the other tests, and testing the same thing twice is going to slow down our e2e runs. I'm fine to make a tech debt card to look into removing duplicate tests later though; if the sharding etc. stuff you're working on speeds things up a lot then double testing is less of a big deal.